numpy/distutils/ccompiler.py

```diff
@@ -620,7 +625,20 @@ def matcher(version_string):
             version = m.group('version')
             return version
 
-    status, output = exec_command(version_cmd, use_tee=0)
+    try:
+        output = subprocess.check_output(version_cmd)
+        output = output.decode(locale.getpreferredencoding(False),
```

Comment:
This stuff will probably be abstracted to a helper function as [Eric suggested in a related PR](https://github.com/numpy/numpy/pull/11900#discussion_r216027013), and probably moved to an `else` block as done there too.